include/roscco/ros_to_oscc.h

```diff
@@ -13,6 +13,52 @@ extern "C" {
 #include <roscco/EnableDisable.h>
 #include <roscco/SteeringCommand.h>
 #include <roscco/ThrottleCommand.h>
+#include <std_msgs/Bool.h>
+
+// struct to enable detection of a rising or falling e_stop flag.
+struct e_stop_status {
+    bool current_e_stop_message{false};
+    bool previous_e_stop_message{false};
+    bool e_stop{false};
+
+    void setPrevious() {
```

Comment:
Instead of `setPrevious`, I think it's better to be a setter for `e_stop`, so maybe `setCurrent`. The caller doesn't need to know about setting the previous, the `setCurrent` will record this automatically after it sets the current value.

Reply:
Yep, I think I was a bit overzealous with my function making.